packages/polling-controller/src/StaticIntervalPollingController.test.ts

```diff
@@ -0,0 +1,242 @@
+
+        expect(controller._executePoll.mock.calls).toMatchObject([
+          ['mainnet', {}],
+          ['sepolia', {}],
+        ]);
+        await advanceTime({ clock, duration: TICK_TIME });
+
+        expect(controller._executePoll.mock.calls).toMatchObject([
+          ['mainnet', {}],
+          ['sepolia', {}],
+          ['mainnet', {}],
+        ]);
+        await advanceTime({ clock, duration: TICK_TIME });
+
+        expect(controller._executePoll.mock.calls).toMatchObject([
+          ['mainnet', {}],
+          ['sepolia', {}],
+          ['mainnet', {}],
+          ['sepolia', {}],
+        ]);
+        await advanceTime({ clock, duration: TICK_TIME });
+
+        expect(controller._executePoll.mock.calls).toMatchObject([
+          ['mainnet', {}],
+          ['sepolia', {}],
+          ['mainnet', {}],
+          ['sepolia', {}],
+          ['mainnet', {}],
+        ]);
+        await advanceTime({ clock, duration: TICK_TIME });
+
+        expect(controller._executePoll.mock.calls).toMatchObject([
+          ['mainnet', {}],
+          ['sepolia', {}],
+          ['mainnet', {}],
+          ['sepolia', {}],
+          ['mainnet', {}],
+          ['sepolia', {}],
+        ]);
+      });
+    });
+  });
+
+  describe('stopPollingByPollingToken', () => {
+    it('should stop polling when called with a valid polling that was the only active pollingToken for a given networkClient', async () => {
+      const pollingToken = controller.startPollingByNetworkClientId('mainnet');
+      await advanceTime({ clock, duration: 0 });
+      expect(controller._executePoll).toHaveBeenCalledTimes(1);
+      await advanceTime({ clock, duration: TICK_TIME });
+      controller.stopPollingByPollingToken(pollingToken);
+      await advanceTime({ clock, duration: TICK_TIME });
+      expect(controller._executePoll).toHaveBeenCalledTimes(2);
+      controller.stopAllPolling();
+    });
+    it('should not stop polling if called with one of multiple active polling tokens for a given networkClient', async () => {
+      const pollingToken1 = controller.startPollingByNetworkClientId('mainnet');
+      await advanceTime({ clock, duration: 0 });
+
+      controller.startPollingByNetworkClientId('mainnet');
+      expect(controller._executePoll).toHaveBeenCalledTimes(1);
+      await advanceTime({ clock, duration: TICK_TIME });
+      controller.stopPollingByPollingToken(pollingToken1);
+      await advanceTime({ clock, duration: TICK_TIME });
+      expect(controller._executePoll).toHaveBeenCalledTimes(3);
+      controller.stopAllPolling();
+    });
+    it('should error if no pollingToken is passed', () => {
+      controller.startPollingByNetworkClientId('mainnet');
+      expect(() => {
+        controller.stopPollingByPollingToken(undefined as unknown as any);
+      }).toThrow('pollingToken required');
+      controller.stopAllPolling();
+    });
+
+    it('should publish "pollingComplete" when stop is called', async () => {
```

Comment:
What are your thoughts on moving this test to a different describe for `onPollingCompleteByNetworkClientId`?

Reply:
yep makes sense!

Reply:
done here: https://github.com/MetaMask/core/pull/3636/commits/823aaa188f572a8aa76973a012ae9bfa8da518e0